python/rmm/_lib/device_buffer.pyx

```diff
@@ -92,39 +97,47 @@ cdef class DeviceBuffer:
         return buf
 
     @staticmethod
-    @cython.boundscheck(False)
-    cdef DeviceBuffer c_frombytes(const unsigned char[::1] b,
+    cdef DeviceBuffer c_to_device(const unsigned char[::1] b,
                                   uintptr_t stream=0):
-        if b is None:
-            raise TypeError(
-                "Argument 'b' has incorrect type"
-                " (expected bytes, got NoneType)"
+        return to_device(b, stream)
+
+    @staticmethod
+    def to_device(const unsigned char[::1] b, uintptr_t stream=0):
+        return to_device(b, stream)
+
+    cpdef copy_to_host(self, unsigned char[::1] hb=None, uintptr_t stream=0):
+        cdef const device_buffer* dbp = self.c_obj.get()
+        cdef size_t s = dbp.size()
+
+        if hb is None:
+            # NumPy leverages huge pages under-the-hood,
+            # which speeds up the copy from device to host.
+            hb = np.empty((s,), dtype="u1")
+        elif len(hb) < s:
+            raise ValueError(
+                "Argument `hb` is too small. Need space for %i bytes." % s
+            )
+        elif len(hb) > s:
+            hb = hb[:s]
+            warnings.warn(
+                "Argument `hb` larger than needed."
+                " Will fill only first %i bytes." % s,
+                RuntimeWarning
```

Comment:
I don't think we need this warning or slicing. Regardless the pointer is to the start of the host buffer, and it's really common to copy data to an allocation larger than the data.

Reply:
Also, slicing is object specific and could potentially return by copy and change our host pointer.

Reply:
> I don't think we need this warning or slicing.

Yeah no strong feelings about the warning.

> Regardless the pointer is to the start of the host buffer, and it's really common to copy data to an allocation larger than the data.

Users could also pick their memoryview to match the right size even if their underlying buffer is larger. If they used a `memoryview` and sliced it themselves, this would take a view (not a copy).

> Also, slicing is object specific and could potentially return by copy and change our host pointer.

In general yes, but this is a specific object. In particular it is a Cython memoryview. It always takes a view when sliced.

Reply:
Dropped the warning 🙂